include/libdnf/transaction/comps_group.hpp

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright (C) 2017-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef LIBDNF_TRANSACTION_COMPS_GROUP_HPP
+#define LIBDNF_TRANSACTION_COMPS_GROUP_HPP
+
+
+#include "transaction_item.hpp"
+
+#include <memory>
+#include <vector>
+
+
+namespace libdnf::transaction {
+
+
+class CompsGroupPackage;
+class Transaction;
+
+
+enum class CompsPackageType : int {
+    CONDITIONAL = 1 << 0,  // a weak dependency
+    DEFAULT = 1 << 1,      // installed by default, but can be unchecked in the UI
+    MANDATORY = 1 << 2,    // always installed
+    OPTIONAL = 1 << 3      // not installed by default, but can be checked in the UI
+};
+
+
+class CompsGroup : public TransactionItem {
+public:
+    explicit CompsGroup(Transaction & trans);
+
+    const std::string & get_group_id() const noexcept { return group_id; }
+    void set_group_id(const std::string & value) { group_id = value; }
+
+    const std::string & get_name() const noexcept { return name; }
+    void set_name(const std::string & value) { name = value; }
+
+    const std::string & get_translated_name() const noexcept { return translated_name; }
+    void set_translated_name(const std::string & value) { translated_name = value; }
+
+    CompsPackageType get_package_types() const noexcept { return package_types; }
+    void set_package_types(CompsPackageType value) { package_types = value; }
+
+    /// Create a new CompsGroupPackage object and return a reference to it.
+    /// The object is owned by the CompsGroup.
+    CompsGroupPackage & new_package();
+
+    /// Get list of packages associated with the group.
+    const std::vector<std::unique_ptr<CompsGroupPackage>> & get_packages() { return packages; }
+
+    //static std::vector< TransactionItemPtr > getTransactionItemsByPattern(
+    //    libdnf::utils::SQLite3Ptr conn,
+    //    const std::string &pattern);
+
+    std::string to_string() const { return get_group_id(); }
+
+private:
+    friend class CompsGroupPackage;
+    std::string group_id;
+    std::string name;
+    std::string translated_name;
+    CompsPackageType package_types;
+    std::vector<std::unique_ptr<CompsGroupPackage>> packages;
+};
+
+
+class CompsGroupPackage {
+public:
+    explicit CompsGroupPackage(CompsGroup & group);
```

Comment:
Why constructor doesn't require CompsPackageType?

Reply:
Because it also doesn't require the other attributes.
An empty object is created and all the attributes are set after.